version: "3.9"
services:
  api:
    user: django-user
    env_file:
      - .env
    build:
      context: ./api
      args:
        - DEV=false
    volumes:
      - ./api/app:/app
      - ./logs:/var/log/live_streaming_list:rw
    ports:
      - "8000:8000"
    environment:
      - DJANGO_SETTINGS_MODULE=config.settings
      - WERKZEUG_DEBUG_PIN=off
      - TZ="Asia/Seoul"
    command: >
      sh -c "mkdir -p /app/logs &&
             python manage.py collectstatic --noinput &&
             python manage.py migrate &&
             gunicorn --bind 0.0.0.0:8000 config.asgi:application --workers 4 --worker-class uvicorn.workers.UvicornWorker --timeout 120 --log-level debug
             --access-logfile /app/logs/access.log --error-logfile /app/logs/error.log"
    networks:
      - app-network

  redis:
    image: "redis:alpine"
    ports:
      - "6379:6379"
    networks:
      - app-network

  celery:
    build:
      context: ./api
    command: sh -c "mkdir -p /app/logs && celery -A config worker --loglevel=info"
    volumes:
      - ./api/app:/app
    env_file:
      - .env
    depends_on:
      - redis
    networks:
      - app-network

networks:
  app-network:
    driver: bridge
